lib/workers/bin/heartbeat_check.rb

```diff
@@ -0,0 +1,50 @@
+require "optparse"
+
+options = {}
+opt_parser = OptionParser.new do |opts|
+  opts.banner = "usage: #{File.basename $PROGRAM_NAME, '.rb'} [HEARTBEAT_FILE]"
+
+  opts.on("-b=HBFILE", "--heartbeat-file=HBFILE", "Heartbeat file to read (overrides arg val)") do |val|
+    options[:heartbeat_file] = val
+  end
+
+  opts.on("-g=GUID", "--guid=GUID", "Use this guid for finding the heartbeat file") do |val|
+    options[:guid] = val
+  end
+
+  opts.on("-v", "--[no-]verbose", "Verbose output") do |val|
+    options[:verbose] = val
+  end
+
+  opts.on("-h", "--help", "Displays this help") do
+    puts opts
+    exit
+  end
+end
+opt_parser.parse!
+
+require File.expand_path("../../miq_defaults.rb", __FILE__)
+heartbeat_file   = options[:heartbeat_file] || ARGV[0]
+heartbeat_file ||= Workers::MiqDefaults.heartbeat_file(options[:guid])
+
+# Determine if we are within the timeout, otherwise exit 1
+if File.exist?(heartbeat_file)
+  require "time"
+  current_time = Time.now.utc
+
+  mtime    = File.mtime(heartbeat_file).utc
+  contents = File.read(heartbeat_file)
+  timeout  = if contents.empty?
+               (mtime + Workers::MiqDefaults.heartbeat_timeout).utc
+             else
+               Time.parse(contents).utc
+             end
+
+  if current_time < timeout
+    puts 0 if options[:verbose]
```

Comment:
Not really, it will just get parsed as the message value. If something isn't json formatted then it will still get recorded it just won't have all the other values.